lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -200,13 +200,13 @@ internal void Reset()
 
         /// <summary>
         /// This method is called when receiving ICompletedTask event during task running or system shutting down.
-        /// Removes the task from running tasks
+        /// Removes the task from running tasks if it was running
         /// Changes the task state from RunningTask to CompletedTask
         /// </summary>
         /// <param name="completedTask"></param>
         internal void RecordCompletedTask(ICompletedTask completedTask)
         {
-            RemoveRunningTask(completedTask.Id);
+            _runningTasks.Remove(completedTask.Id);
```

Comment:
RemoveRunningTask requires the task was in _runningTasks, meaning the task must be running before.
This method can be called in the scenario when driver sends event for the task to close, and the task returns from Call() gracefully, resulting driver receives ICompletedTask. In this case, the task was not in _runningTasks. Using _runningTasks.remove() fits the case even when the task is not in the collection.